src/Client.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace Utopia\Fetch;
+
+/**
+ * Client class
+ * @package Utopia\Fetch
+ */
+class Client
+{
+    public const METHOD_GET = 'GET';
+    public const METHOD_POST = 'POST';
+    public const METHOD_PUT = 'PUT';
+    public const METHOD_PATCH = 'PATCH';
+    public const METHOD_DELETE = 'DELETE';
+    public const METHOD_HEAD = 'HEAD';
+    public const METHOD_OPTIONS = 'OPTIONS';
+    public const METHOD_CONNECT = 'CONNECT';
+    public const METHOD_TRACE = 'TRACE';
+    /**
+     * Flatten request body array to PHP multiple format
+     *
+     * @param array<mixed> $data
+     * @param string $prefix
+     * @return array<mixed>
+     */
+    private function flatten(array $data, string $prefix = ''): array
+    {
+        $output = [];
+        foreach ($data as $key => $value) {
+            $finalKey = $prefix ? "{$prefix}[{$key}]" : $key;
+
+            if (is_array($value)) {
+                $output += $this->flatten($value, $finalKey); // @todo: handle name collision here if needed
+            } else {
+                $output[$finalKey] = $value;
+            }
+        }
+
+        return $output;
+    }
+  /**
+     * Private static method to process the data before making the request
+     * @param array<string, string> $headers
+     * @param array<string>|array<string, mixed> $body
+     * @param array<string, mixed> $query
+     * @param string $method
+     * @param string $url
+     */
+    private function processData(
+        array &$headers,
+        array &$body,
+        array $query,
+        string &$method,
+        string &$url
+    ): void {
+        if(!$method) { // if method is not set, set it to GET by default
+            $method = self::METHOD_GET;
+        } else { // else convert the method to uppercase
+            $method = strtoupper($method);
+        }
+        // if there are no headers but a body set header to json by default
+        if(!isset($headers['content-type']) && is_array($body) && count($body) > 0) {
+            $headers['content-type'] = 'application/json';
+        }
+        if(isset($headers['content-type'])) {
+            switch($headers['content-type']) { // Convert the body to the appropriate format
+                case 'application/json':
+                    $body = json_encode($body);
+                    break;
+                case 'application/x-www-form-urlencoded':
+                    $body = $this->flatten($body);
+                    break;
+                case 'multipart/form-data':
+                    $body = $this->flatten($body);
+                    break;
+                case 'application/graphql':
+                    $body = $body[0];
+                    break;
+                default:
+            }
```

Comment:
you could replace that using `match`, is gonna be more elegant
https://www.php.net/manual/en/control-structures.match.php